### PR TITLE
Support unknown diagnostic contexts better

### PR DESCRIPTION
**Pull request by brycelelbach**
If we don't recognize the diagnostic we've been given, and thus have no way to extract source locations and find relevant code, we should still try to help by sending the diagnostic to the AI. This PR makes some changes to better support this.

### CODE REVIEW

File: src/cwhy/cwhy.py [hunk 1]
```diff
@@ -149,32 +155,28 @@ async def complete(user_prompt):
...
-            This is my error:
+            """
+        user_prompt += f"""
+        This is my error:
 
-            {ctx.abridged_diagnostic}
+        {ctx.abridged_diagnostic}
+        """
+        if fix:
+            user_prompt = f"""
```
Comment (nicovank) [inline src/cwhy/cwhy.py:177]:
```suggestion
            user_prompt += f"""
```

File: src/cwhy/cwhy.py [hunk 2]
```diff
@@ -115,8 +116,13 @@ def __init__(self, diagnostic):
...
               max_code_locations -= 1
 
+        # If the diagnostic didn't come from a context that we know about and
+        # handle in the above loop, we should make sure it's not too long.
+        if not self.code_locations and line == len(diagnostic_lines) - 1:
+            line = min(line, 50)
+
         self.unabridged_diagnostic = '\n'.join(diagnostic_lines) + '\n'
-        self.abridged_diagnostic = '\n'.join(diagnostic_lines[0:line]) + '\n'
+        self.abridged_diagnostic = '```\n' + '\n'.join(diagnostic_lines[0:line]) + '\n```\n'
```
Comment (nicovank) [inline src/cwhy/cwhy.py:125]: Extreme nit on previous code.

```suggestion
        self.abridged_diagnostic = '```\n' + '\n'.join(diagnostic_lines[:line]) + '\n```\n'
```

File: src/cwhy/cwhy.py [hunk 3]
```diff
@@ -149,32 +155,28 @@ async def complete(user_prompt):
...
-
-            This is my error:
-
-            {ctx.abridged_diagnostic}
 
-            Suggest code to fix the problem. Surround the code in backticks (```).
-            """
-        else:
-            user_prompt = f"""
+        user_prompt = str()
```
Comment (nicovank) [inline src/cwhy/cwhy.py:163]:
```suggestion
        user_prompt = ""
```

**Review by nicovank — COMMENTED** [review]
Couple nits, one problem (`+=` vs `=`).

If we don't find code, maybe we also should not include "This is my code:" with nothing following it in the prompt. I can make the change when this is merged.

**Review by emeryberger — APPROVED** [review]
LGTM

**Pull request merged**